caproto/_data.py

```diff
@@ -212,7 +212,11 @@ def __init__(self, *, alarm=None, value=None, timestamp=None,
         self.reported_record_type = reported_record_type
 
         if self._max_length is None:
-            self._max_length = self.calculate_length(value)
+            # Use the current length as maximum, if unspecified.
+            self._max_length = max(self.calculate_length(value), 1)
```

Comment:
would it make more sense to blow up and refuse to to a 0 length input without also providing a max length?

Reply:
I would like that better.

Reply:
Meh, I have changed my mind about whether it is worth breaking currently-deployed IOCs over this. I think, "You didn't give me a length or a value, so I will assume this is a scalar," is acceptable.

Reply:
Maybe starting issuing warnings, though.

Reply:
Final answer? ;)